CRM/Core/BAO/EntityTag.php

```diff
@@ -15,9 +15,15 @@
  * @package CRM
  * @copyright CiviCRM LLC https://civicrm.org/licensing
  */
-class CRM_Core_BAO_EntityTag extends CRM_Core_DAO_EntityTag {
+class CRM_Core_BAO_EntityTag extends CRM_Core_DAO_EntityTag implements \Civi\Core\Service\AutoSubscriber {
```

Comment:
To my eye, this is the only example where `class AutoSubscriber` wouldn't work (ie where `interface AutoSubscriber` is more mixable) -- but I think it's more confusing this way:

* When someone says `new CRM_Core_BAO_EntityTag()`, are they preparing a DB-operation or a container-service? They are very different things. 
* You can already mix-in `EventSubscriberInterface` here. I don't see what you're gaining from it.
* There are now two overlapping/concurrent paths to interpreting the `EventSubscriberInterface` on this class. In one path, all BAO objects are allowed to subscribe with static methods. In another path, the service-instance is allowed to subscribe with non-static methods. Both paths use the same `getSubscribedEvents()`. Observe duplicate registrations:

    ```
    $ cv debug:event-dispatcher /preDel/
    The debug command ignores the container cache.
    [Event] civi.dao.preDelete
    +-------+------------------------------------------------------------------------------+
    | Order | Callable                                                                     |
    +-------+------------------------------------------------------------------------------+
    | #1    | CRM_Core_BAO_EntityTag::preDeleteOtherEntity($e)                             |
    | #2    | civirules_trigger_predelete(\$e)                                             |
    | #3    | $(CRM_Core_BAO_EntityTag)->preDeleteOtherEntity($e) [CRM_Core_BAO_EntityTag] |
    +-------+------------------------------------------------------------------------------+
    ```